crates/sui-types/src/messages_checkpoint.rs

```diff
@@ -220,58 +213,68 @@ impl CheckpointSummary {
 
 impl BcsSignable for CheckpointSummary {}
 
-// TODO: Rename SignedCheckpoint to SignedCheckpointSummary
 #[derive(Clone, Debug, Serialize, Deserialize)]
-pub struct SignedCheckpoint {
-    pub checkpoint: CheckpointSummary,
-    pub authority: AuthorityName,
-    signature: AuthoritySignature,
+pub struct CheckpointSummaryEnvelope<S> {
+    pub summary: CheckpointSummary,
+    pub auth_signature: S,
 }
 
-impl SignedCheckpoint {
+pub type SignedCheckpointSummary = CheckpointSummaryEnvelope<AuthoritySignInfo>;
+
+impl SignedCheckpointSummary {
     /// Create a new signed checkpoint proposal for this authority
     pub fn new(
         sequence_number: CheckpointSequenceNumber,
         authority: AuthorityName,
         signer: &dyn signature::Signer<AuthoritySignature>,
         transactions: &CheckpointContents,
         previous_digest: Option<CheckpointDigest>,
-    ) -> SignedCheckpoint {
+    ) -> SignedCheckpointSummary {
         let checkpoint = CheckpointSummary::new(sequence_number, transactions, previous_digest);
-        SignedCheckpoint::new_from_summary(checkpoint, authority, signer)
+        SignedCheckpointSummary::new_from_summary(checkpoint, authority, signer)
     }
 
     pub fn new_from_summary(
         checkpoint: CheckpointSummary,
         authority: AuthorityName,
         signer: &dyn signature::Signer<AuthoritySignature>,
-    ) -> SignedCheckpoint {
+    ) -> SignedCheckpointSummary {
         let signature = AuthoritySignature::new(&checkpoint, signer);
 
-        SignedCheckpoint {
-            checkpoint,
-            authority,
-            signature,
+        SignedCheckpointSummary {
+            summary: checkpoint,
+            auth_signature: AuthoritySignInfo {
+                // TODO: Add proper epoch info
+                epoch: 0,
+                authority,
+                signature,
+            },
         }
     }
 
+    pub fn authority(&self) -> &AuthorityName {
+        &self.auth_signature.authority
+    }
+
     /// Checks that the signature on the digest is correct
     pub fn verify(&self) -> Result<(), SuiError> {
```

Comment:
No entirely sure where this would best happen, but I think it needs to happen: the signature is now associated with an epoch, and the checkpoint will also be associated with an epoch. We need to ensure / check the two are the same as part of verification.

Reply:
Does checkpoint need to be associated with an epoch, or is it sufficient to have its signature to contain the epoch?